app/routes/_gcn.circulars/circulars.server.ts

```diff
@@ -46,6 +54,28 @@ export const getDynamoDBAutoIncrement = memoizee(
   { promise: true }
 )
 
+export const getDynamoDBVersionAutoIncrement = memoizee(async function (
+  circularId: number
+) {
+  const db = await tables()
+  const doc = db._doc as unknown as DynamoDBDocument
+  const counterTableName = db.name('circulars')
+  const tableName = db.name('circulars_history')
+  const dangerously =
+    (await (db._db as unknown as DynamoDB).config.endpoint?.())?.hostname ==
+    'localhost'
+
+  return new DynamoDBHistoryAutoIncrement({
+    doc,
+    counterTableName,
+    counterTableKey: { circularId },
+    attributeName: 'version',
+    tableName,
+    initialValue: 1,
+    dangerously,
+  })
+})
```

Comment:
This will store a memoized autoincrement object _for each circular ID_. That's too much!

Reply:
Please _don't_ memoize this.

Reply:
Yup! updated in latest commit